packages/cw-orch-core/src/contract/interface_traits.rs

```diff
@@ -188,12 +192,19 @@ pub trait CwOrchQuery<Chain: QueryHandler + ChainState>:
         query_map: Map<'a, K, T>,
         key: K,
     ) -> Result<T, CwEnvError> {
-        self.get_chain()
+        self.environment()
             .wasm_querier()
             .map_query(self.address()?, query_map, key)
     }
 }
 
+impl<Chain: ChainState, T: ?Sized + ContractInstance<Chain>> Environment<Chain> for T {
+    fn environment(&self) -> &Chain {
+        #[allow(deprecated)]
+        self.get_chain()
```

Comment:
kind of weird to use a deprecated function here. 

If we sunset this fn in the trait we should also sunset it in the `Contract` type.

So update the old fn: 
```rust
fn get_chain(&self) -> &Chain {
        self.as_instance().environment()
}
```

And then have the same code here: 

```rust
fn get_chain(&self) -> &Chain {
        self.as_instance().environment()
}
```

And implement Environment for Contract.

Reply:
I can't implement Environment for Contract, because of this compiler error : 
 `downstream crates may implement trait 'contract::interface_traits::ContractInstance<_>' for type 'contract::contract_instance::Contract<_>'`

So I decided to just have a function in the impl rather than implementing the trait